src/mcp-server/azext_mcp/server.py

```diff
@@ -0,0 +1,200 @@
+            return build_command_group_help_info(command_name, command_group, self.command_table, self.group_table)
+
+        return 'Command or group not found'
+
+    def get_command_arguments_schema(self, command_name: str):
+        """Get argument help and schema for a specific command."""
+        # Implementation here
+        command = self.command_table.get(command_name)
+        if not command:
+            return None
+        if hasattr(command, '_args_schema'):
+            return handle_arg_schema(command._args_schema)
+        help_info = build_command_help_info(command_name, command)
+        if help_info and 'arguments' in help_info:
+            return handle_help_schema(help_info)
+        return None
+
+    def invoke_command(self, command_name: str, arguments: dict | None = None):
+        """Invoke a command with JSON-described arguments."""
+        from azure.cli.core.commands import LongRunningOperation, _is_poller, _is_paged, AzCliCommandInvoker
+
+        command = self.command_table.get(command_name)
+        if not command:
+            return None
+        if arguments is None:
+            arguments = {}
+        arguments = {"cmd": command, **arguments}  # Ensure 'cmd' is passed to the command
+        result = command(arguments)
+        transform_op = command.command_kwargs.get('transform', None)
+        if transform_op:
+            result = transform_op(result)
+
+        if _is_poller(result):
+            result = LongRunningOperation(command.cli_ctx, 'Starting {}'.format(command.name))(result)
+        elif _is_paged(result):
+            result = list(result)
+
+        from azure.cli.core.util import todict
+        result = todict(result, AzCliCommandInvoker.remove_additional_prop_layer)
+        return {'result': result}
+
+
+class AzMCP(FastMCP):
+
+    def __init__(
+            self,
+            cli_ctx: AzCli,
+            name: str | None = None,
+            instructions: str | None = None,
+            event_store: EventStore | None = None,
+            *,
+            tools: list[Tool] | None = None):
+        super().__init__(
+            name or "AZ MCP",
+            instructions,
+            event_store,
+            tools=tools)
+        self.cli_ctx = cli_ctx
+        self.az_cli_bridge = AzCLIBridge(self.cli_ctx)
+        self._register_primitives()
+        # self._register_resources()
+    
+    def _register_primitives(self):
+        super().resource(
+            "az://{command_name_path}",
+            name="Azure CLI Command Help",
+            description="Retrieve comprehensive help documentation for Azure CLI commands and command groups. "
+                   "Provides detailed information including command syntax, parameters, examples, and usage patterns. "
+                   "Path format: 'az://command/subcommand' (e.g., 'az://login', 'az://vm/create', 'az://storage/account/list')",
+            mime_type="application/json",
+        )(self.command_help_resource)
```

Comment:
This doesn't work because the implemetation of MCP python package has only very basic support of RFG 6570. https://github.com/modelcontextprotocol/python-sdk/issues/378